### PR TITLE
MTDSA-855 - Make tax year mandatory for calculation.

### PR DESCRIPTION
**Pull request by hicnar**

### CODE REVIEW

File: conf/application.conf [hunk 1]
```diff
@@ -204,13 +204,13 @@ controllers {
...
 
   uk.gov.hmrc.selfassessmentapi.resources.CannedCalculationResource = {
-    needsAuth = true
-    needsLogging = true
-    needsAuditing = true
-    authParams = {
-      account = ""
-      pattern = "/ni/(\\w+)/.*"
-    }
+    needsAuth = false
```
Comment (GlassAndOneHalf) [inline conf/application.conf]: I think these should all be set to true. @rakeshavasarala ?

Comment (rakeshavasarala) [inline conf/application.conf]: yes

File: test/uk/gov/hmrc/selfassessmentapi/resources/Jsons.scala [hunk 2]
```diff
@@ -535,6 +535,24 @@ object Jsons {
...
+    def request() : JsValue = {
+      Json.parse(
+        s"""
+          |{
+          |  "taxYear": "2016-17"
+          |}
+         """.stripMargin)
+    }
+
+    def emptyRequest() : JsValue = {
```
Comment (GlassAndOneHalf) [inline test/uk/gov/hmrc/selfassessmentapi/resources/Jsons.scala]: You can use `Json.obj()` here to construct an empty object.

File: conf/application.conf [hunk 3]
```diff
@@ -204,13 +204,13 @@ controllers {
...
+    needsLogging = false
+    needsAuditing = false
+//    authParams = {
```
Comment (GlassAndOneHalf) [inline conf/application.conf]: This will also need to be uncommented

File: test/uk/gov/hmrc/selfassessmentapi/resources/Jsons.scala (hunk 2 above)
Comment (hicnar) [inline test/uk/gov/hmrc/selfassessmentapi/resources/Jsons.scala]: Applied

File: conf/application.conf (hunk 1 above)
Comment (hicnar) [inline conf/application.conf]: fixed

**Comment by GlassAndOneHalf** [comment]
LGTM!

**Pull request merged**